src/pyhf/infer/calculators.py

```diff
@@ -168,3 +169,160 @@ def _false_case():
+
+        Args:
+            data (`tensor`): The observed data.
+            pdf (~pyhf.pdf.Model): The statistical model adhering to the schema ``model.json``.
+            init_pars (`tensor`): The initial parameter values to be used for fitting.
+            par_bounds (`tensor`): The parameter value bounds to be used for fitting.
+            ntoys (`int`): Number of toys to use (how many times to sample the underlying distributions)
+            track_progress (`bool`): Whether to display the `tqdm` progress bar or not (outputs to `stderr`)
+
+        Returns:
+            ~pyhf.infer.calculators.ToyCalculator: The calculator for toy-based quantities.
+
+        """
+        self.ntoys = ntoys
+        self.data = data
+        self.pdf = pdf
+        self.init_pars = init_pars or pdf.config.suggested_init()
+        self.par_bounds = par_bounds or pdf.config.suggested_bounds()
+        self.track_progress = track_progress
+
+    def distributions(self, poi_test, track_progress=None):
+        """
+        Probability Distributions of the test statistic value under the signal + background and background-only hypothesis.
+
+        Args:
+            poi_test: The value for the parameter of interest.
+            track_progress (`bool`): Whether to display the `tqdm` progress bar or not (outputs to `stderr`)
+
+        Returns:
+            Tuple (~pyhf.infer.calculators.EmpiricalDistribution): The distributions under the hypotheses.
+
+        """
+        tensorlib, _ = get_backend()
+        sample_shape = (self.ntoys,)
+
+        signal_pars = self.pdf.config.suggested_init()
+        signal_pars[self.pdf.config.poi_index] = poi_test
+        signal_pdf = self.pdf.make_pdf(tensorlib.astensor(signal_pars))
+        signal_sample = signal_pdf.sample(sample_shape)
+
+        bkg_pars = self.pdf.config.suggested_init()
+        bkg_pars[self.pdf.config.poi_index] = 0.0
+        bkg_pdf = self.pdf.make_pdf(tensorlib.astensor(bkg_pars))
+        bkg_sample = bkg_pdf.sample(sample_shape)
+
+        tqdm_options = dict(
+            total=self.ntoys,
+            leave=False,
+            disable=not (
+                track_progress if track_progress is not None else self.track_progress
+            ),
+            unit='toy',
+        )
+
+        signal_qtilde = []
+        for sample in tqdm.tqdm(signal_sample, **tqdm_options, desc='Signal-like'):
+            signal_qtilde.append(
+                qmu(poi_test, sample, self.pdf, signal_pars, self.par_bounds)
+            )
+        signal_qtilde = tensorlib.astensor(signal_qtilde)
+
+        bkg_qtilde = []
+        for sample in tqdm.tqdm(bkg_sample, **tqdm_options, desc='Background-like'):
+            bkg_qtilde.append(
+                qmu(poi_test, sample, self.pdf, bkg_pars, self.par_bounds)
+            )
+        bkg_qtilde = tensorlib.astensor(bkg_qtilde)
+
+        s_plus_b = EmpiricalDistribution(signal_qtilde)
+        b_only = EmpiricalDistribution(bkg_qtilde)
```

Comment:
```suggestion
        signal_test_stat = []
        for sample in tqdm.tqdm(signal_sample, **tqdm_options, desc='Signal-like'):
            signal_test_stat.append(
                qmu(poi_test, sample, self.pdf, signal_pars, self.par_bounds)
            )

        bkg_test_stat = []
        for sample in tqdm.tqdm(bkg_sample, **tqdm_options, desc='Background-like'):
            bkg_test_stat.append(
                qmu(poi_test, sample, self.pdf, bkg_pars, self.par_bounds)
            )

        s_plus_b = EmpiricalDistribution(tensorlib.astensor(signal_test_stat))
        b_only = EmpiricalDistribution(tensorlib.astensor(bkg_test_stat))
```

I think that we should use use the more generic "test statistic" (`test_stat`) here rather than `qtilde` as that is making an assumption about the test being done. Seems like it could lead to confusion later and that a more explicit example in the docstring that uses `qtilde` would be more enlightening.